### DISCUSSION TITLE
Windows + WSL: Run Vivado to build openwifi FPGA (openwifi-hw)

### DISCUSSION THREAD

**Discussion by JiaoXianjun** [open-sdr/openwifi#341] [Show and tell]
Our main working environment is Ubuntu LTS, such as version 20, 22. If you have the same/similar environment,
you can skip this document and just follow those tips about OS in places of other documents.

If your main working environment is Windows, here are some tips.
- It is possible that use Windows + WSL for FPGA development.
  - I have built FPGA bitstream successfully by using Windows 10 + WSL 2 + Vivado (installed in WSL 2) with some **necessary tricks**.
- WSL supports Linux GUI program running smoothly in it with GUI displayed in Windows.
  - You can find many materials on this topic.
  - You can try to install a Linux GUI program, such as gedit, in WSL to verify this easily..
- After the Linux GUI program can run in your WSL, you can download Vivado Linux installer from Xilinx and run it in WSL.
  - Install the Vivado with the installer GUI program. Just as how it does in any other OS.
- After Vivado is installed successfully, you might need the following two tricks for Viterbi Decoder (Evaluation) license
and memory size
- Normally the Viterbi Decoder license needs to be bound to a HOSTID (MAC address of a NIC). The problem is that the MAC address of WSL will change randomly after close/re-open. You can run the following bash command line-by-line in WSL to set a fixed MAC address.
  ```
  sudo ip link add name bond0 type bond mode active-backup
  wantmac=aa:bb:cc:dd:ee:ff
  mac=$(ip link show bond0 | awk '/ether/ {print $2}')
  if [[ $mac !=  $wantmac ]]; then
  sudo ip link set dev bond0 address $wantmac
  fi
  ```
- If WSL does not have enough memory and swap size, Vivado might quit abnormally while building the FPGA project. You can assign
WSL bigger memory and swap size by creating a .wslconfig file for WSL.
  - For example, memory=12GB and swap=16GB work well in my case.
  - Follow this link: https://learn.microsoft.com/en-us/windows/wsl/wsl-config#example-wslconfig-file

**Comment by MatiasMontanna** [comment]
Hello! Can you explain if it's possible to replace the default Wi-Fi driver (for example: Realtek) with openwifi? If yes, how? And, it's possible to use openwifi as a driver in Windows 11?

There's benefits of using this instead of default driver?

Thanks for reading.
